src/main/java/net/neoforged/neoforge/common/NeoForgeEventHandler.java

```diff
@@ -92,6 +105,38 @@ public void tagsUpdated(TagsUpdatedEvent event) {
         if (event.shouldUpdateStaticData()) {
             CommonHooks.updateBurns();
         }
+        if (event.getUpdateCause() == TagsUpdatedEvent.UpdateCause.SERVER_DATA_LOAD) {
+            DATA_MAPS.apply();
+        }
+    }
+
+    @SubscribeEvent
+    public void onDpSync(final OnDatapackSyncEvent event) {
+        final List<ServerPlayer> players = event.getPlayer() == null ? event.getPlayerList().getPlayers() : List.of(event.getPlayer());
+        RegistryManager.getDataMaps().forEach((registry, values) -> {
+            final var regOpt = event.getPlayerList().getServer().overworld().registryAccess()
+                    .registry(registry);
+            if (regOpt.isEmpty()) return;
+            players.forEach(player -> {
+                if (player.connection.isVanillaConnection()) {
+                    return;
+                }
+                final var playerMaps = player.connection.connection.channel().attr(RegistryManager.ATTRIBUTE_KNOWN_DATA_MAPS).get();
```

Comment:
> Or is it possible that different players have a different set of known data maps?
That is indeed possible if a data map is optionally synced. I don't think such distributor would make sense as it would both be a bit ugly and too niche for an API.